crates/sui-framework/packages/deepbook/sources/clob.move

```diff
@@ -390,13 +390,15 @@ module deepbook::clob {
                     emit_order_canceled<BaseAsset, QuoteAsset>(pool_id, maker_order);
                 } else {
                     // Calculate how much quote asset (maker_quote_quantity) is required, including the commission, to fill the maker order.
-                    let (flag, maker_quote_quantity) = clob_math::mul_round(maker_base_quantity, maker_order.price);
+                    let (flag, maker_quote_quantity) = clob_math::safe_mul_round(
+                        maker_base_quantity,
+                        maker_order.price
+                    );
                     if (flag) maker_quote_quantity = maker_quote_quantity + 1;
-                    (flag, maker_quote_quantity) = clob_math::mul_round(
+                    maker_quote_quantity = clob_math::safe_mul(
                         maker_quote_quantity,
                         FLOAT_SCALING + pool.taker_fee_rate
                     );
-                    if (flag) maker_quote_quantity = maker_quote_quantity + 1;
```

Comment:
Can you please add this change (and the test that failed before the change + passes after) in a separate PR instead of in this one? We should avoid semantics-preserving refactoring and behavior changes.

Reply:
Adopt!

Reply:
Looks like the genesis snapshot needs to be updated; cargo test; and cargo insta review in sui-genesis.

For this, 
My original plan was to update the snapshot after perfecting and testing all the codes that have not been covered,
And squash into one commit to prepare for merge.